x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/PartiallyCachedShardAllocationIntegTests.java

```diff
@@ -106,7 +108,7 @@ private MountSearchableSnapshotRequest prepareMountRequest() throws InterruptedE
     public void testPartialSearchableSnapshotNotAllocatedToNodesWithoutCache() throws Exception {
         final MountSearchableSnapshotRequest req = prepareMountRequest();
         final RestoreSnapshotResponse restoreSnapshotResponse = client().execute(MountSearchableSnapshotAction.INSTANCE, req).get();
-        assertThat(restoreSnapshotResponse.getRestoreInfo().successfulShards(), equalTo(0));
+        assertThat(Optional.ofNullable(restoreSnapshotResponse.getRestoreInfo()).map(RestoreInfo::successfulShards).orElse(0), equalTo(0));
```

Comment:
This changes the meaning of the test doesn't it? Why are we getting a `null` restore info here? We should only be getting that if we had a master failover in the middle of operations here shouldn't we?

Reply:
I think it can be null if a restore is not completed for all shards: https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/snapshots/RestoreService.java#L1389

Reply:
Yes but it should be completed right away because all shards should fail to restore?

Reply:
But I agree it's weird, we don't check it for null almost everywhere in the snapshot tests.

Reply:
@original-brownbear  I looked at it deeper, all the test failures happen due to the cluster turning RED via `desiredBalance`
```
[2023-02-21T22:36:59,495][INFO ][o.e.c.r.a.AllocationService] [node_s0] current.health="RED" message="Cluster health status changed from [YELLOW] to [RED] (reason: [reconcile-desired-balance])." previous.health="YELLOW" reason="reconcile-desired-balance"
```

I believe the issued got fixed in #93635. This is also corroborated by the fact that the test fails only in 7.6 and 7.7.

Reply:
I'm not sure if need this null check or it would be better to add an `ensureGreen` assertion before mounting a snapshot